### DISCUSSION TITLE
Peridigm can't open ".g" mesh file

### DISCUSSION THREAD

**Discussion by guangdongzhang** [peridigm/peridigm#240] [Q&A]
I use the Cubit export the mesh file as ".g" and select NETCDF4/5, but when I run the Peridigm, it shows that can't open the ".g" file. How do you export the mesh file from Cubit? What's the correct setting?

**Comment by djlittl** [comment]
The file format should be “exodus”.

**Comment by guangdongzhang** [comment]
Why the given examples are ".g" file for mesh?


| |
张广栋
|
|
***@***.***
|
---- Replied Message ----
| From | David ***@***.***> |
| Date | 7/21/2022 21:22 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [peridigm/peridigm] Peridigm can't open ".g" mesh file (Discussion #240) |

The file format should be “exodus”.

**Comment by djlittl** [comment]
Because “.g” stands for genesis, which the folks who wrote this I/O library use to denote an input file.  For whatever reason, they choose to use references from the Bible for the names of input and output files.  Genesis (.g) is creation, and exodus (.e) means departure.

**Reply by djlittl** [reply]
A genesis file (.g) is the same as an exodus file (.e) except that genesis files typically don’t have any data fields on them.  The underlying file structure is the same.

**Reply by sahirbutt** [reply]
That was informative, now the names 'genesis' and 'exodus' make much more sense to me... 🙂

**Reply by djlittl** [reply]
It’s not every day that the answer to a technical question is in the Bible!

**Comment by guangdongzhang** [comment]
So, for us to use the Peridigm, only ".e" mesh file can be used?


| |
张广栋
|
|
***@***.***
|
---- Replied Message ----
| From | David ***@***.***> |
| Date | 7/21/2022 21:36 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [peridigm/peridigm] Peridigm can't open ".g" mesh file (Discussion #240) |

Because “.g” stands for genesis, which the folks who wrote this I/O library use to denote an input file. For whatever reason, they choose to use references from the Bible for the names of input and output files. Genesis (.g) is creation, and exodus (.e) means departure.

**Reply by sahirbutt** [reply]
No, the input to peridigm is a (*.g) mesh file and the output from peridigm is a (*.e) file. Both of which belong to the exodus format.
Use Cubit to export the (*.g) file, with export type "Exodus".

**Comment by guangdongzhang** [comment]
Okay, so the export ".g" file with exodus format can be used directly in Peridigm, right?


| |
张广栋
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 7/21/2022 21:49 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | Re: [peridigm/peridigm] Peridigm can't open ".g" mesh file (Discussion #240) |

No, the input to peridigm is a (.g) mesh file and the output from peridigm is a (.e) file. Both of which belong to the exodus format.
Use Cubit to export the (*.g) file, with export type "Exodus".

**Reply by sahirbutt** [reply]
Yes.
